packages/fields/src/types/CloudinaryImage/views/blocks/single-image.js

```diff
@@ -0,0 +1,213 @@
+    return {
+      float: 'right',
+      marginLeft: '10px',
+      width: '50%',
+    };
+  } else {
+    return {
+      display: 'block',
+      margin: '0px auto',
+      width: '100%',
+    };
+  }
+}
+export function Node(props) {
+  let alignment = props.node.data.get('alignment');
+  return (
+    <figure
+      css={{ display: 'flex', flexDirection: 'column', ...getImageStyle(alignment) }}
+      {...props.attributes}
+    >
+      <image.ImageAlignmentContext.Provider
+        value={useMemo(() => {
+          return {
+            alignment,
+            onAlignmentChange(newAlignment) {
+              props.editor.setNodeByKey(props.node.key, {
+                data: props.node.data.set('alignment', newAlignment),
+              });
+            },
+          };
+        }, [props.node.key, alignment, props.editor, props.node.data])}
+      >
+        {props.children}
+      </image.ImageAlignmentContext.Provider>
+    </figure>
+  );
+}
+
+export let schema = {
+  nodes: [
+    {
+      match: [{ type: image.type }],
+      min: 1,
+      max: 1,
+    },
+    {
+      match: [{ type: caption.type }],
+      min: 1,
+      max: 1,
+    },
+  ],
+  normalize(editor, error) {
+    switch (error.code) {
+      case 'child_min_invalid': {
+        if (error.index === 0) {
+          // the image has been deleted so we also want to delete the image-container
+          editor.removeNodeByKey(error.node.key);
+        }
+        if (error.index === 1) {
+          // the caption has been deleted
+          // the user probably doesn't want to delete the image
+          // they probably just wanted to remove everything in the caption
+          // so the caption gets removed,  we insert another caption
+          editor.insertNodeByKey(error.node.key, 1, Block.create('caption'));
+        }
+        return;
+      }
+      case 'node_data_invalid': {
+        if (error.key === 'alignment') {
+          editor.setNodeByKey(error.node.key, {
+            data: error.node.data.set('alignment', 'center'),
+          });
+        }
+        return;
+      }
+    }
+    console.log(error);
+  },
+  data: {
+    alignment(value) {
+      switch (value) {
+        case 'center':
+        case 'left':
+        case 'right': {
+          return true;
+        }
+      }
+      return false;
+    },
+  },
+};
+
+export let plugins = [
+  insertImages({
+    extensions: imageExtensions,
+    insertImage: insertImageBlockFromFile,
+  }),
+  {
+    onDragStart(event, editor, next) {
+      const { value } = editor;
+      const { document } = value;
+      const node = findNode(event.target, editor);
+      if (node.type === image.type) {
+        const ancestors = document.getAncestors(node.key);
+        let imgContainer = ancestors.get(ancestors.size - 1);
+        if (imgContainer.type === type) {
+          editor.moveToRangeOfNode(imgContainer);
+        }
+      }
+
+      next();
+    },
+  },
+];
+
+export function serialize({ node }) {
```

Comment:
Each _Block_ handles its own serialisation (executed before sending the slate.js JSON to the server).

In this block, it looks for a nested `image` block which contains the `file` that needs uploading, and removes that `file` field from the `data` object so it doesn't get accidentally serialised and stored as JSON. Instead, it adds it as part of a `CloudinaryImage` mutation.